### PR TITLE
Add documentation for SAML Keystone Federation

### PR DESCRIPTION
**Pull request by edtubillara**
This is the first inital commit for adding customer documentation on how to
use SAML federation.

### CODE REVIEW

**Comment by edtubillara** [comment]
I will need Nikhil and NIthya to review...

**Comment by LeslieLundquist** [comment]
OK, let me know when you're ready to merge. Thanks!

**Comment by LeslieLundquist** [comment]
@edtubillara Isn't there already a document with this title? Is it meant to replace the older one? (We just did some type of update to that one for 3.1 so I would be surprised...)

**Comment by edtubillara** [comment]
Hi, this document is closely related to Keystone to Keystone federation. It is similar because it is SAML federation.

**Comment by LeslieLundquist** [comment]
@edtubillara good to know, thanks! In the "title" metadata at the top of the file, could we add SAML somewhere in that title? That's the one that will show for users. It is exactly the same title as the previous document, so that will show up twice in the list, and it will be confusing for them.

**Comment by edtubillara** [comment]
Leslie thanks for catching that. I have updated it and am still waiting on Nikhil and Nithya for review.

File: _keystone/saml-federation.md [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+**Please note that any administrative commands on this page requires
+the user to have `cloud_admin` access.
+
+## <a name="integrating_idp_and_keystone"></a>Integrating the IDP and Keystone
+A Blue Box operator will need to work with a `Company XYZ IDP` integration
+engineer to setup both systems to work together.
+
+A Blue Box operator will need the following:
+* IDP Metadata URL Endpoint
+* IDP entity ID from the IDP Metadata
```
Comment (lihkin213) [inline _keystone/saml-federation.md]: What if there is no metadata URL? Are there specific things that we should require the IdP to provide in lieu of the metadata URL?

File: _keystone/saml-federation.md [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+The name given is what the name is inside the SAML assertion. The id is what will
+be used in the Keystone mappings.
+
+After the Blue Box operator is finished integrating `Company XYZ IDP` with
+Keystone, a mapping called `mapping-for-company-xyz-idp` should be
+created (or a similar mapping name with the IDP name in it).
+
+The steps for managing the mapping is:
+
+**1. First, manage groups (Federated users will ahve the same access rights as the group**
```
Comment (lihkin213) [inline _keystone/saml-federation.md]: Think the `First` and `next` are redundant as you have them as numbered steps

File: _keystone/saml-federation.md [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+
+
+def get_project_and_federated_session():
+    # authenticate to the idp
+    saml_auth = SAMLPassword(auth_url=OS_AUTH_URL,
+                             identity_provider=OS_IDENTITY_PROVIDER,
+                             identity_provider_url=OS_IDENTITY_PROVIDER_URL,
+                             username=OS_USERNAME,
+                             password=OS_PASSWORD,
+                             protocol=OS_PROTOCOL)
```
Comment (lihkin213) [inline _keystone/saml-federation.md:468]: The formatting above this line is messed up.

Comment (edtubillara) [reply _keystone/saml-federation.md:468]: I have updated the formatting for other parts of the page. Also, the page appearing on github greatly differs on what it looks like in the end. There is an explanation on how to run a local server here: https://github.com/IBM-Blue-Box-Help/help-documentation/pull/164

File: _keystone/saml-federation.md (hunk 2 above)
Comment (edtubillara) [reply _keystone/saml-federation.md]: Done

File: _keystone/saml-federation.md (hunk 1 above)
Comment (edtubillara) [reply _keystone/saml-federation.md]: I will add a note that says we currently only support having a metadata URL.

File: _keystone/saml-federation.md [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+**Please note that any administrative commands on this page requires
+the user to have `cloud_admin` access.**
+
+## <a name="integrating_idp_and_keystone"></a>Integrating the IDP and Keystone
+A Blue Box operator will need to work with a `Company XYZ IDP` integration
+engineer to setup both systems to work together.
+
+A Blue Box operator will need the following:
+
+ * IDP Metadata URL Endpoint (We currently only support retrieving Metadata through an URL and not a ile)
```
Comment (lihkin213) [inline _keystone/saml-federation.md]: incorrect spelling, should be `file` not `ile`

**Comment by LeslieLundquist** [comment]
Line 74 has a typo, the word should be "have" instead of "ahve"

**Comment by edtubillara** [comment]
Thanks Leslie, I have updated the PR.

File: _keystone/saml-federation.md (hunk 2 above)
Comment (narengan) [reply _keystone/saml-federation.md]: 👍

**Pull request merged**